img_cmd_training_new_dataset.py

```diff
@@ -0,0 +1,280 @@
+from argparse import ArgumentParser
+
+import matplotlib.pyplot as plt
+import os.path
+from os import path
+
+from fastai.vision.all import *
+from fastai.callback.progress import CSVLogger
+
+import wandb
+
+# Initialize W&B run (if not already initialized)
+run = wandb.init(
+    project="third-testing",
+    entity="vuhachau2412",
+    notes="A set of small/useless datasets for testing.",
+    job_type="dataset-upload"
+)
+# Load the dataset artifact
+artifact = run.use_artifact("arcslaboratory/Multirun-testing-1K+/larger-perfect-dataset:v0")
+artifact_dir = artifact.download()
+
+
+dataset_path = artifact_dir + '/'   # Path to the extracted images from the artifact
+
+from torch.utils.data import Dataset
+
+# Constants 
+# Load the dataset artifact
+
+VALID_PCT = 0.05
+NUM_REPLICATES = 2
+NUM_EPOCHS = 1
+# DATASET_DIR = Path("/data/clark/summer2021/datasets")
+DATASET_DIR = Path(dataset_path)
+MODEL_PATH_REL_TO_DATASET = Path("cmd_models_fixed")
+DATA_PATH_REL_TO_DATASET = Path("cmd_data_fixed")
+# VALID_MAZE_DIR = Path("../Mazes/validation_mazes8x8/")
+
+compared_models = {
+    "resnet18": resnet18,
+    "resnet34": resnet34
+}
+
+DATASET_DIR.ls()
+
+def filename_to_class(filename: str) -> str:
+    angle = float(filename.split("_")[1].split(".")[0].replace("p", "."))
+    if angle > 0:
```

Comment:
Instead of `> 0` we should come up with a threshold.

Reply:
We are not quite sure we understand what you mean by threshold with the angle. Would you mind elaborating?

Reply:
This requires two changes:

* In `boxnav` we should use the angle in the filenames without any thresholding.
* Here we should say that any angle between, for example, -5 degrees and 5 degrees is given the label `forward.